openreview/venue/process/invite_assignment_pre_process.js

```diff
@@ -27,6 +28,53 @@ async function process(client, edge, invitation) {
   const profiles = await client.tools.getProfiles([edge.tail], true)
   const userProfile = profiles[0]
 
+  // Check for complete profile, if no profile then go to pending sign up
+  if (profileReqs && !userProfile.id.includes('@')) {
```

Comment:
I wonder if we can move this code to the tools.py so it can be used from the preprocess and the cron job.

Reply:
Ok! I can move it. To allow this JS preprocess to use it, should it get added here?: https://github.com/openreview/openreview-js/blob/8c7cd7f370459391a9cb4cc416223a386c287c6b/packages/client/src/tools.js

Reply:
you check by profile id instead of email address like `userProfile.id.startswith("~")`

Reply:
let's move this to the tools.py so the process function is not so large.

Reply:
But this is a JS preprocess so it would need to go in the JS version. What do you think?

Reply:
yes, let's do that

Reply:
This has been moved to tools.js: https://github.com/openreview/openreview-js/pull/72